File: app/src/main/java/com/yatatsu/passiveviewsample/data/api/GitHubService.java
package com.yatatsu.passiveviewsample.data.api;

import com.yatatsu.passiveviewsample.data.model.Repository;
import com.yatatsu.passiveviewsample.data.model.User;

import java.util.List;

import retrofit.http.GET;
import retrofit.http.Path;
import rx.Observable;


public interface GitHubService {

    @GET("/users/{username}")
    Observable<User> getUser(
            @Path("username") String username
    );

    @GET("/users/{username}/repos")
    Observable<List<Repository>> getRepositories(
            @Path("username") String username
    );
}

File: app/src/main/java/com/yatatsu/passiveviewsample/data/api/GitHubService.kt
package com.yatatsu.passiveviewsample.data.api

import com.yatatsu.passiveviewsample.data.model.Repository
import com.yatatsu.passiveviewsample.data.model.User

import retrofit.http.GET
import retrofit.http.Path
import rx.Observable


interface GitHubService {

  @GET("/users/{username}")
  fun getUser(
      @Path("username") username: String): Observable<User>

  @GET("/users/{username}/repos")
  fun getRepositories(
      @Path("username") username: String): Observable<List<Repository>>
}
